capstone/src/main/webapp/lesson-loader.js

```diff
@@ -6,6 +6,8 @@ const VIDEO = "video";
 const CONTENT = "content";
 const IMAGE = "image";
 
+const queryString = "/lesson?room_id=" + getRoomId() + "&action=" + getJoinStatus();
```

Comment:
Terminology-wise, the query string in a url is purely the part after the '?' (and before the '#' if there is one), not the whole path. I would just call this like lessonPath or something. Good call pulling this out though

Reply:
To make this more robust, I would use the JS classes around URLs (this includes for all requests in the app). You can write this as:

```
lessonEndpoint = new URL("/lesson", window.location.href);
lessonEndpoint.searchParams.set(ROOM_ID, getRoomId());
lessonEndpoint.searchParams.set(ACTION, getJoinAction());
```
then later when you want to use it you can fetch this instead of a string `fetch(locationEndpoint, {method...})`

Reply:
gotcha that works perfectly